presto-spi/src/main/java/io/prestosql/spi/security/Identity.java

```diff
@@ -76,13 +76,15 @@ public boolean equals(Object o)
             return false;
         }
         Identity identity = (Identity) o;
-        return Objects.equals(user, identity.user);
+        return Objects.equals(user, identity.user) &&
+                Objects.equals(principal, identity.principal) &&
+                Objects.equals(roles, identity.roles);
```

Comment:
This has far-reaching implications.
Should `Identity(MrX, someRoles)` be considered different than `Identity(MrX, emptyMap())`?
Should `Identity(MrX, kerberosPrincipal)` be considered different than `Identity(MrX)` (eg the view creator)?
I think the answer is -- _it depends_.

For this reason, I think that `Identity` has no "obvious equality semantics" and I would rather see it not implement `equals()`.